tests/base.py

```diff
@@ -43,6 +43,7 @@ def setUpClass(cls):
                 cls.all_rules = rc.rules
                 cls.rule_lookup = rc.id_map
                 cls.production_rules = rc.filter(production_filter)
+                cls.bbr = rc.get_bbr()
```

Comment:
I tried that initially, it will not work because the `building_block_type` is not metadata, it is in the rule contents. 

Additionally, since it is only in the rule contents for building block rules as opposed to all rules, adding a filter rule contents method like below will also not work, as the `building_block_type` is missing in non-building block rules and will cause the function to fail. 

```python
def contents_filter(**metadata) -> Callable[[TOMLRule], bool]:
    """Get a filter callback based off rule metadata"""
    flt = dict_filter(metadata)

    def callback(rule: TOMLRule) -> bool:
        target_dict = rule.contents.data.to_dict()
        return flt(target_dict)

    return callback


```

Reply:
@terrancedejesus and I also discussed filtering on tags as a way of doing this, but we ultimately decided that it would add quite a bit of additional overhead processing that would largely be unnecessary.

Reply:
That's a shame. Makes a lot sense though.